python/lsst/verify/extract_metricvalues.py

```diff
@@ -0,0 +1,243 @@
+    """Print all metrics with measured values in the given repo.
+
+    Parameters
+    ----------
+    butler : `lsst.daf.butler.Butler`
+        Butler to load values from.
+    kind : `str`
+        Kind of metrics to load.
+    data_id_keys : `tuple` [`str`], optional
+        List of Butler dataId keys to restrict the printed output to;
+        for example: ``("detector", "visit")``.
+    data_id_restriction : `dict`, optional
+        Only include values whose dataId matches these key:value pairs;
+        for example: ``{"detector": 50}``.
+    verbose : `bool`, optional
+        Print extra information when loading values.
+
+    Returns
+    -------
+    output : `str`
+        A formatted string with all the requested metric values.
+    """
+    def match_data_id(data_id):
+        """Return True if ``data_id`` matches ``data_id_restriction``.
+        """
+        for key, value in data_id_restriction.items():
+            if not (data_id[key] == value):
+                return False
+        return True
+
+    def value_formatter(value):
+        return f"{value}"
+
+    match kind:
+        case "value":
+            result = load_value(butler, verbose=verbose)
+        case "timing":
+            result = load_timing(butler, verbose=verbose)
+
+            def value_formatter(value):
+                return f"{value.datum.label}: {value.quantity:.4}"
+        case "memory":
+            result = load_memory(butler, verbose=verbose)
+
+            def value_formatter(value):
+                return f"{value.datum.label}: {value.quantity.to(u.Mbyte):.5}"
+        case _:
+            raise RuntimeError(f"Cannot handle kind={kind}")
+
+    for (data_id, metric), value in sorted(result.items()):
+        if data_id_restriction is not None and not match_data_id(data_id):
+            continue
+
+        print(f"{_data_id_label(data_id, data_id_keys)};", value_formatter(value))
+
+
+def print_diff_metrics(butler1, butler2, data_id_keys=None, verbose=None):
+    """Load metric values from two repos and print their differences.
+
+    This only supports differencing metrics that aren't time or memory-related.
+
+    Parameters
+    ----------
+    butler1, butler2 : `lsst.daf.butler.Butler`
+        Butlers to load values to difference from.
+    data_id_keys : `tuple` [`str`], optional
+        List of Butler dataId keys to restrict the printed output to;
+        for example: ``("detector", "visit")``.
+    verbose : `bool`, optional
+        Print extra information when loading values, and about failures.
+    """
+    result1 = load_value(butler1)
+    result2 = load_value(butler2)
+
+    same = 0
+    failed = 0
+    for key in result1:
+        data_id, metric = key
+        try:
+            value1 = result1[key]
+            value2 = result2[key]
+        except KeyError:
+            print(f"Result 2 does not contain metric '{metric}'' for dataId '{data_id}'")
+            failed += 1
+            continue
```

Comment:
Yeah, I decided to punt on some of the error handling, because I only have a single use case for this right now. I don't know that it's worth expanding that until we have some more uses?